Marketing Tools: Refactor to use new components

pablinos opened the pull request:
#### Changes proposed in this Pull Request

This change updates the marketing tools page to use the new components created for the earn page.

#### Testing instructions

Use calypso.live and go to `/marketing/tools`. ~~Currently, both versions are shown on the page for comparison and the CSS needs updating.~~ This has been updated so the PR is ready for review. The styling has changed slightly, but is in line with the new `/earn/` section, which also uses the same pattern.

The focus now (alongside the usual code checks) would be to ensure that the functionality remains the same at that the styling changes are acceptable.

Fixes #34531

pablinos committed:
Marketing Tools: Update to use the new PromoSection component

After reworking the Earn page, we had a set of generic components for
creating promo pages with this kind of design pattern. As a result it
seemed like we should refactor the Marketing Tools section to use them.

This is WIP!

pablinos committed:
First pass refactor of the marketing tools page.

It should be functionally complete, but requires CSS updates.

Commit:
CSS updates to marketing section after code refactor

Preview: https://cl.ly/2f668010e781

Commit:
Update spaces to tabs

Addressing indentation issue @eliorivero mentioned here:
https://github.com/Automattic/wp-calypso/commit/76b31dfc541bde70af933b4a
5bd5bdf7fddf0e6f#r34536949

pablinos commented on client/my-sites/marketing/style.scss:
```diff
@@ -77,6 +77,24 @@
+			max-width: 64px;
+		}
+		.action-panel__body p {
+			flex-grow: 0;
```
I think this was set to `flex-grow: 1` so that the CTA buttons get anchored to the bottom of the card. Is it related to the Google Ads card text that we need to turn that off?

pablinos replied:
I've just noticed, that I didn't mention you in this comment @davemart-in.  Any thoughts on this?

davemart-in replied:
Yeah, I intentionally changed that. The space between the content and the button felt too extreme in some of the cards.

pablinos replied:
That's good, I was just checking it was intended!

pablinos committed:
Updated to remove the previous version of the marketing tools

The previous version of the marketing tools section was left in so that
the styling and functionality could be easily compared with this
refcator. In order to make it ready for review I've removed it, along
with the associated imports.

cbauerman reviewed (changes requested):
First off: sorry for the delay! This Pr is looking really good, outside of one change that was a strongly felt requirement from design: the button narrow and on the lower left.

<img width="533" alt="Screen Shot 2019-08-20 at 5 15 06 PM" src="https://user-images.githubusercontent.com/2810519/63393086-5ae09f80-c36e-11e9-90b0-79e6052f8733.png">

It looks like we have the potential for a Marketing Tools specific css override, would that be possible to get those buttons in similar style to the original? Thanks!

pablinos commented:
Thanks @cbauerman! No problem, I forgot to add the 'Needs Review' label too, so it probably didn't notify any of the reviewers I added properly.

I've updated this so the buttons are compact and right-aligned. It's got me wondering about the reasoning for it. If that performs better from a UX point of view, then perhaps we should change the styling of the Earn section. Was this A/B tested?

@huguesvincent @jancavan - Should we run an A/B test on the Earn section in order to see if this button layout is better?

pablinos commented:
Ok, that's good, thanks @jancavan! I'll back out the last commit and that should mean this is good to go. What do you think @cbauerman?

cbauerman reviewed (approved):
Looks great 👍  LGTM!

pablinos committed:
Reset primary card background to white

pablinos commented:
Thanks for looking at this both of you.

> I would also just revert back to the card with the white background.

I think that @davemart-in got some feedback to use the background to further highlight the primary card, but I've reset it to white.

I've also reverted the buttons to their normal size and left-aligned them.

@cbauerman & @jancavan - One last look at this would be really appreciated.

cbauerman reviewed (approved):
Changes make buttons look consistent w/ `my-plan` buttons. Everything looks great, LGTM!

jancavan commented:
I think we should just hide the illustration in the header once it hits `<1040px`. So essentially, retaining the current behavior. It looks off as the screen gets narrower:

![Screenshot 2019-08-28 21 53 01](https://user-images.githubusercontent.com/4924246/63911237-77b84b00-c9de-11e9-9294-08a57a62edd4.png)

Other than that, LGTM.

pablinos commented:
This is very old and represents a previous view of the marketing page. I'm going to close it.
